src/expr/skolem_manager.cpp

```diff
@@ -230,7 +158,20 @@ Node SkolemManager::mkSkolemFunctionTyped(SkolemFunId id,
       // we use @ as a prefix, which follows the SMT-LIB standard indicating
       // internal symbols starting with @ or . are reserved for internal use.
       std::stringstream ss;
-      ss << "@" << id;
+      if (id == SkolemFunId::INTERNAL)
+      {
```

Comment:
I don't know why this case needs special treatment. Can you add documentation for it?

Reply:
Done.